### ISSUE TITLE
PHP 7

### ISSUE THREAD

**Issue by Pimmal** [lansuite/lansuite#322]
When there will be a PHP 7.1 release?
Because my hoster is removing PHP5.6 soon in case of no more updates.

The actual 4.2-beta is not running with PHP7 and the actuall trunk is not ready for production use or?

**Comment by andygrunwald** [comment]
Hey @Pimmal,

we are scheduling a release / collecting TODOs for the next release in #224.
We have a couple of PRs open where we need put more effort in like #83 and #313. Those are related to the new release.
If you find some time to deliver also feedback to open PRs, this would be really appreciated.

> [...] and the actuall trunk is not ready for production use or?

Lets say this: We are not aware of any fatal errors or heavy bugs that are not listed as issues already.
If we find a fatal error or a user reports this, we are fixing those very quick to keep the current master version stable as possible.

We can't name a specific date for the release, because we work on this in our private time and don't earn money with it. Because of this, often the private life and higher priorities take over.
If you want to contribute and join the team, let us know.
We would be happy to answer every open question and onboard you. With more people we would have more progress on the open tasks.

A few questions out of curiosity:

* Which hoster are you using? (to know who is removing PHP 5.6 support)
* Would you be open to add your LAN-Party-Website to #312 and provide us input for *Who is using LANSuite?*?

**Comment by Pimmal** [comment]
> Which hoster are you using? (to know who is removing PHP 5.6 support)
I´m moving to my own webhost server, and there i dont wont to implement PHP 5.6 till December 2018 because there is End of live cycle. (And if i iplement it i will remove it on that date)

**Comment by Pimmal** [comment]
http://lansuite.netsession.net/index.php?mod=picgallery&action=hitlist

> Fehler: 
PHP Warning: mysqli_free_result() expects parameter 1 to be mysqli_result, null given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 322
PHP Warning: mysqli_free_result() expects parameter 1 to be mysqli_result, null given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 322
PHP Warning: mysqli_free_result() expects parameter 1 to be mysqli_result, null given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 322
PHP Warning: mysqli_free_result() expects parameter 1 to be mysqli_result, null given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 322

**Comment by Pimmal** [comment]
http://lansuite.netsession.net/index.php?mod=install&action=ls_conf

> Fatal error: Uncaught Error: Call to a member function get_tag_content() on null in /www/htdocs/w01237f1/netsession.net/lansuite/modules/install/ls_conf.php:79 Stack trace: #0 /www/htdocs/w01237f1/netsession.net/lansuite/index_module.inc.php(76): include_once() #1 /www/htdocs/w01237f1/netsession.net/lansuite/index.php(419): include_once('/www/htdocs/w01...') #2 {main} thrown in /www/htdocs/w01237f1/netsession.net/lansuite/modules/install/ls_conf.php on line 79

**Comment by andygrunwald** [comment]
Thanks for the bugs.
Are you using the latest master already? Or which detail version is this what you have installed?

**Comment by Pimmal** [comment]
> Are you using the latest master already? Or which detail version is this what you have installed?

I have cloned the master branch 10 minutes ago.

**Comment by andygrunwald** [comment]
Thanks for the quick response. We try to fix those quite quickly.

Which PHP-Version + MySQL-Version you are using for your tests?

**Comment by Pimmal** [comment]
> Which PHP-Version + MySQL-Version you are using for your tests?

PHP 7.1 because 7.0 has also end of life in december 2018
Mysql Version: 5.6.38

**Comment by M4LuZ** [comment]
I'm wondering if it would make sense to create an intermediate release that just contains the rework for PHP 7 and maybe some other cherry-picked fixes.
As I do not see that we create a release within the next 1-2 months given the amount of issues that keep turning up after all the refactoring work and the still open issue of the update process itself.

**Comment by Pimmal** [comment]
@M4LuZ therefore i have said i´m testing with PHP 7.1 ;-)

**Comment by andygrunwald** [comment]
> As I do not see that we create a release within the next 1-2 months given the amount of issues that keep turning up after all the refactoring work and the still open issue of the update process itself.

It is not only the refactoring. It is an old system. E.g. all database related errors were there all the time because the queries were not touched. 
I try to focus on the issues that we have defined in #224 to get it out of the door quick.

**Comment by Pimmal** [comment]
@andygrunwald would be cool if the bugs here can be fixed, because then i dont need to install php5.6 on my produvtion server.

i have not such good php knowledge to help here i can only build up easy html design templates ;-)

**Comment by andygrunwald** [comment]
> @andygrunwald would be cool if the bugs here can be fixed, because then i dont need to install php5.6 on my produvtion server.

Will try to get it done quick.
I will ping you here, once the reported issues are fixed.

> i have not such good php knowledge to help here i can only build up easy html design templates ;-)

Contribution is not only writing source code. Actions like

* Improve the documentation
* reporting tickets and help to reproduce them
* testing new features / pull requests with cloning/installing
* Fixing typos
* Translating texts into another language
* Welcome newcomers
* Helping out with support in the issue tracker
* Talking about LANSuite at events like meetups or LANParties
* And similar activities

are always welcome and helps us a lot.

**Comment by M4LuZ** [comment]
> It is not only the refactoring. It is an old system. E.g. all database related errors were there all the time because the queries were not touched.

The difference is that the refactoring so far caused fatal errors due to visibility or inclusion issues, while the DB issues are warnings that can be safely ignored as that is how LS works.

Even if: We have multiple changes that either require a large manual intervention or a yet-to-be-developed upgrade process.

It shouldn't require a lot of time to prepare and buys us a lot of time to get a completely reworked version out of the door. Testing effort - compared to what is still in front of us - is also minimal.

Thus the proposal.

**Comment by Pimmal** [comment]
> Fehler: 
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255

I think that are non critical errors, can i hide them?

**Comment by andygrunwald** [comment]
@Pimmal Can you also post the link here?
The error

```
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
```

marks a database error. Those errors have often a callstack and the root cause is not in `inc/Classes/DB.php on line 255`. The root cause happens before.
A link offers us to see which module and action is affected.
We might also introduce a proper debug mode with call stack.

**Comment by Pimmal** [comment]
http://lansuite.netsession.net/index.php?mod=guestlist&action=details&userid=1
`Fatal error: Uncaught ArgumentCountError: Too few arguments to function LanSuite\Display::FetchIcon(), 1 passed in /www/htdocs/w01237f1/netsession.net/lansuite/modules/hardware/plugins/usrmgr_details_tab.php on line 10 and at least 2 expected in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/Display.php:1224 Stack trace: #0 /www/htdocs/w01237f1/netsession.net/lansuite/modules/hardware/plugins/usrmgr_details_tab.php(10): LanSuite\Display->FetchIcon('cpu') #1 /www/htdocs/w01237f1/netsession.net/lansuite/modules/usrmgr/details.php(403): include_once('/www/htdocs/w01...') #2 /www/htdocs/w01237f1/netsession.net/lansuite/modules/guestlist/details.php(3): include_once('/www/htdocs/w01...') #3 /www/htdocs/w01237f1/netsession.net/lansuite/index_module.inc.php(90): include_once('/www/htdocs/w01...') #4 /www/htdocs/w01237f1/netsession.net/lansuite/index.php(419): include_once('/www/htdocs/w01...') #5 {main} thrown in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/Display.php on line 1224`

http://lansuite.netsession.net/index.php?mod=foodcenter
`Fehler: 
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255
PHP Warning: mysqli_num_rows() expects parameter 1 to be mysqli_result, array given in /www/htdocs/w01237f1/netsession.net/lansuite/inc/Classes/DB.php on line 255`

**Comment by Pimmal** [comment]
http://lansuite.netsession.net/index.php?mod=usrmgr&action=party&party_id=6&user_id=1&mf_step=2&mf_id=1#MF1
`Fatal error: Uncaught Error: Call to a member function IncrementNumber() on null in /www/htdocs/w01237f1/netsession.net/lansuite/modules/usrmgr/party.php:83 Stack trace: #0 /www/htdocs/w01237f1/netsession.net/lansuite/index_module.inc.php(90): include_once() #1 /www/htdocs/w01237f1/netsession.net/lansuite/index.php(419): include_once('/www/htdocs/w01...') #2 {main} thrown in /www/htdocs/w01237f1/netsession.net/lansuite/modules/usrmgr/party.php on line 83`

**Comment by andygrunwald** [comment]
Hey @Pimmal,
I went through the bugs you have reported and fixed them.
Checkout #326, #328, #329, #330 and #332.

Furthermore, we implemented a better debugging mode in #325.
You can now enable the `debugmode`. Go to your config in `inc/base/config.php` and set `debugmode = 1`.

If you have this enabled and an error occurs, the stack trace helps us to fix those faster :)
Once all changes are merged, would you mind to test the fixes?

**Comment by Pimmal** [comment]
http://lansuite.netsession.net/ wenn man einfach nur auf die Seite geht
`Hinweis: 
Dieser Benutzer existiert nicht in unserer Datenbank. Bitte prüfe die eingegebene Email/ID `

**Comment by M4LuZ** [comment]
> http://lansuite.netsession.net/ go to the site with no logged on user.

I just did that and saw no error message in that regard. Where did you see it?

> Mark me if you need testing somewere.

Happy to do that.
Just merged the rework of the tournament module and PayPal functionality.
You can give that a try if you like ;)

**Comment by Pimmal** [comment]
![1](https://user-images.githubusercontent.com/20731374/41157620-b0626ebc-6b26-11e8-9af8-709d5b798813.PNG)
i can give you access if needed

**Comment by andygrunwald** [comment]
@Pimmal Can you try to clear your cookies and destroy the session with this?
Often this is the cause for this (this is what I saw in my environments).

**Comment by Pimmal** [comment]
ok solved, in private tab no problem.

**Comment by Pimmal** [comment]
If you press on the guestlist sho all fo then the design is switched away from "netsession" to "osX"
http://lansuite.netsession.net/index.php?mod=guestlist&action=guestlist&EntsPerPage=0&design=&msExport=

On the usermanager the red flag with adding somebody to party is missing.
http://lansuite.netsession.net/index.php?mod=usrmgr
![1](https://user-images.githubusercontent.com/20731374/41167944-b74e228e-6b44-11e8-8fea-4031f335e4ba.PNG)

When you are logged in the activity overview is missing, it´s only shown select party
![2](https://user-images.githubusercontent.com/20731374/41167948-beda8eac-6b44-11e8-874e-204417cc93f1.PNG)

If you create a new account you get a welcome email, a password email but the account is not created in the Database.
![3](https://user-images.githubusercontent.com/20731374/41168423-34a3a1ae-6b46-11e8-91fd-b33b2734e56f.PNG)

**Comment by Pimmal** [comment]
@M4LuZ you can reopen the party creation without error message, now i have also the same problem witzh crating my 2nd party,

**Comment by aj-gh** [comment]
Quite some PHP7 issues were fixed recently.
For example the mentioned "guestlist show all" design bug was fixed in #412.
I don't seem to be able to reproduce the user signup database issue or the missing flag.

@Pimmal, do you still see any issues with the current Master?
If not I'd suggest to close this issue as it's rather old and several bugs were addressed meanwhile.

**Comment by M4LuZ** [comment]
Closing this as no feedback in regards to open issues has been provided in.....ermmm....three years
